18-temperature-and-humidity-monitor/firmware/README.md

```diff
@@ -0,0 +1,142 @@
+* [PlatformIO](https://platformio.org/)
+
+## Notehub Setup
+
+Sign up for a free account on [notehub.io](https://notehub.io) and [create a new project](https://dev.blues.io/quickstart/notecard-quickstart/notecard-and-notecarrier-pi/#set-up-notehub).
+
+## Hardware Setup
+
+1. Assemble Notecard and Notecarrier as described [here](https://dev.blues.io/quickstart/notecard-quickstart/notecard-and-notecarrier-f/).
+2. Plug the Swan into the Notecarrier, aligning the Swan's male headers with the Notecarrier's female headers.
+3. (OPTIONAL) The BME280 breakout has an LED to indicate that its powered on. This draws a significant amount of current, especially compared to the sensor itself. To maximize battery life, you can disconnect this LED by [cutting the JP1 jumper trace](https://learn.sparkfun.com/tutorials/how-to-work-with-jumper-pads-and-pcb-traces/cutting-a-trace-between-jumper-pads) (see [the schematic](https://cdn.sparkfun.com/assets/0/9/6/b/4/Qwiic_BME280_Schematic_attempt2.pdf)) on the back of the board with a hobby knife.
+4. Plug one end of the Qwiic cable into one of the Notecarrier Qwiic ports and the other end into one of the Qwiic ports on the BME280 breakout board.
+5. Plug the LiPo battery's JST cable into the Notecarrier port labeled "LIPO".
+6. Connect the micro USB cable from your development PC to the Swan.
+
+## Notecard Firmware Setup
+
+The Notecard should use [firmware version 3.5.1](https://dev.blues.io/notecard/notecard-firmware-updates/#v3-5-1-october-7th-2022) or higher. The simplest way to update firmware is to do an [over-the-air update](https://dev.blues.io/notecard/notecard-firmware-updates/#ota-dfu-with-notehub).
+
+## Swan Firmware Setup
+
+1. Open Visual Studio Code.
+2. Click the PlatformIO icon on the left hand side and open this "firmware" folder with Quick Access > PIO Home > Open > Open Project.
+3. Open the file src/main.cpp. Uncomment this line
+
+```c
+#define PRODUCT_UID "com.your-company:your-product-name"
+```
+
+and replace `com.your-company:your-product-name` with your ProductUID from [Notehub Setup](#notehub-setup).
+
+4. Click the PlatformIO icon on the left hand side again and click "Build" under Project Tasks > bw_swan_r5 > General. You should see "SUCCESS" in the terminal output pane. The firmware image is now ready to be flashed to the Swan.
+5. Press and hold the button labeled "BOOT" on the Swan, and, while holding BOOT, press and release the button labeled "RST", and finally release the BOOT button. The Swan is now ready to be flashed.
+6. Click "Upload" under Project Tasks > bw_swan_r5 > General. You should see "SUCCESS" in the terminal output pane.
+
+## Operation
+
+At this point, the Swan firmware should be running and you should start to see events show up on Notehub under your project's "Events" view. There are 3 important Notefiles.
+
+### `data.qo`
+
+By default, the Swan will publish temperature and humidity data from the BME280 to this file every 3 minutes. Ex:
```

Comment:
Is `Ex:` a normal way to write Example? I would prefer something like

> For example:

> A typical event looks like this:

>  "E.g." is "For example"

Probably too terse and formal for our style, but just mentioning it. (Although having said that, it does feature on dev.blues.io.)

Reply:
I'm an e.g. fan. I'll change to that.